### PR TITLE
Meganetic Lockwave + Multi choice option for CHOOSEMULTIZONE

### PR DESCRIPTION
**Pull request by KuKetto**
now you can add optional multi choice to MZ logic (CHOOSEMULTIZONE and MAYCHOOSEMULTIZONE)
if you want to add a multichoice you need to use `PREPENDLASTRESULT` to add maximum and minimum choice count to mz input

example
```php
AddDecisionQueue("MULTIZONEINDICES", $otherPlayer, "MYCHAR:type=E");
AddDecisionQueue("PREPENDLASTRESULT", $otherPlayer, "MAXCOUNT-" . 3 . ",MINCOUNT-" . 2 . ",");
```

Meganetic Lockwave should work as intended. It autoblocks for the opponent if the equipment hasn't blocked before. Breaking the chain works also, does not clear the effect

### CODE REVIEW

File: CoreLogic.php
Comment (OotTheMonk) [inline CoreLogic.php:1]: This change of adding it to the combat chain during evaluatecombatchain gives me pause; this is called a lot of times, and times outside of ProcessInput where game actions are taken, including from spectators. What I would recommend instead is add validation when they try to pass blocks without blocking with that equipment
image

**Review by OotTheMonk — CHANGES_REQUESTED** [review]
Added comment regarding the evaluatecombatchain change

File: CoreLogic.php
Comment (KuKetto) [reply CoreLogic.php:1]: My thought on it was that only adds 1 additional check to regular cases. And in the turn where the card is play we could a little bit more of execution time. 
However of course, I understand your worries. I will update this today.

**Comment by KuKetto** [comment]
Made the change as suggested. I haven't placed it into `ContinueDecisionQueue`. It would be hard to undo the block pass or would require reverting the game state. Maybe if we will have more forced blocking effects during a turn we can add a function there to make `PassInput` a bit more cleaner. 
image

**Review by OotTheMonk — APPROVED** [review]

**Pull request merged**